packages/itwinui-react/src/core/LabeledInput/LabeledInput.tsx

```diff
@@ -104,24 +104,24 @@ export const LabeledInput = React.forwardRef((props, ref) => {
         </Label>
       )}
 
-      <InputWithDecorations
+      <InputFlexContainer
```

Comment:
Simply swapping in `InputFlexContainer` does not do anything; `InputWithDecorations` already uses `InputFlexContainer` underneath. The important thing to realize is that these two components are designed for complex/arbitrary custom decorations on either side ([example](https://itwin.github.io/iTwinUI/react/?story=searchbox--basic-with-custom-items&theme=dark)).

In the case of `LabeledInput` and `LabeledTextarea`, we only allow one simple svg to passed into `svgIcon` and we know exactly where to put it, so we can use a simpler approach similar to v2:
1. show the input/textarea as-is.
2. absolutely position the icon on the right end
3. add some padding onto the input/textarea.

This will also fix the strange bug where the textarea resize handle appears offset from the bottom-right corner.

| v2 | v3 |
| --- | --- |
| ![image](https://github.com/iTwin/iTwinUI/assets/9084735/3902225c-e8be-41ca-809c-6bb70c74a9b4) | ![image](https://github.com/iTwin/iTwinUI/assets/9084735/71dfaf97-50d8-4222-8946-b3ea1258dd3d) |

Reply:
To confirm, is [`d3b16d5` (#1753)](https://github.com/iTwin/iTwinUI/pull/1753/commits/d3b16d516d08191ef413e57bbad6cdbbb2ea2d7f) (WIP) a step in the right direction? In this commit, I tried re-using `iui-input-with-icon`.

This fixed the textarea resize issue you mentioned. But I still need to investigate why the border color is not the same as the status color when a `LabeledInput`/`LabeledTextarea` is focused.

https://github.com/iTwin/iTwinUI/assets/45748283/7375163b-ec8b-4f6e-a7eb-8cca1d7bee55

Reply:
Yeah, that's generally the right direction. Do we already have the `iui-input-with-icon` class? i.e. no css changes needed?

Reply:
> Do we already have the `iui-input-with-icon` class? i.e. no css changes needed?

Yes, I believe it is used in Select and the ComboBox to show the down arrow icon ([full repo search](https://github.com/search?q=repo%3AiTwin%2FiTwinUI%20iui-input-with-icon&type=code)).

By reusing `iui-input-with-icon`, big CSS changes may not be needed, but I still might need to make a few small changes to fix issues such as the border color issue. Will investigate this further.